source/rust_verify/src/rust_to_vir_expr.rs

```diff
@@ -780,6 +835,199 @@ pub(crate) fn pattern_to_vir<'tcx>(
     Ok(pattern)
 }
 
```

Comment:
Please consider moving the following (till 1030) to a new file, e.g. `rust_to_vir_invariant.rs`: it's already factored out, I believe, so it should mostly be a matter of copying over the imports (if it's more cumbersome than I think, let me know!).

Reply:
i don't mind too much, but I do feel like it belongs in this file (it is part of the large mutual recursion on expressions & statements)

Reply:
Makes sense!